Release Zebra v1.1.0 (#7259)
ZcashFoundation/zebra: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org).

## [Zebra 1.1.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.1.0) - 2023-07-18

> Zebra now uses less memory after the final checkpoint by deduplicating note commitment trees in the non-finalized state.

Reviewer: Optional: Is this a noticeable amount of memory?

I think we're talking about less than a kilobyte per note commitment tree, for 3 shielded protocols, for 100 blocks (so less than 1MB). It's also possible we were already de-duplicating the trees by cloning the `Arc` rather than the whole tree.

Author: > Is this a noticeable amount of memory?

I don't think so, but I wasn't sure.

> It's also possible we were already de-duplicating the trees by cloning the Arc rather than the whole tree.

That's correct, unless updating trees with new note commitments can reproduce the same tree.

Reviewer: > > It's also possible we were already de-duplicating the trees by cloning the Arc rather than the whole tree.
> 
> That's correct, unless updating trees with new note commitments can reproduce the same tree.

No, it's not possible to add a commitment and get the same tree. Adding a new note commitment always changes the tree, because it always adds a leaf, and sometimes removes a subtree.

> Zebra now uses less memory after the final checkpoint by deduplicating note commitment trees in the non-finalized state.

Author: ```suggestion
```

> This release also adds new mempool metrics, fixes panics when cancelling tasks on shutdown, and improves the usability of Zebra's Docker images (particularly for mining).

Author: ```suggestion
This release adds new mempool metrics, fixes panics when cancelling tasks on shutdown, detects subcommand name typos on the command-line, and improves the usability of Zebra's Docker images (particularly for mining).
```

### Breaking Changes

> - Zebra now detects subcommand name typos on the command-line. If you want to give Zebra a list of tracing filters, use `zebrad start --filters debug,...` ([#7056](https://github.com/ZcashFoundation/zebra/pull/7056))

Reviewer: Optional: This might be worth a brief mention in the intro, because it might impact a few users.

Author: Added in https://github.com/ZcashFoundation/zebra/pull/7259/commits/8186212f4bfc5af1311aa99ba8d57abbb80f550d.

### Security

- Avoid initiating outbound handshakes with IPs for which Zebra already has an active peer ([#7029](https://github.com/ZcashFoundation/zebra/pull/7029))
- Rate-limit inbound connections per IP ([#7041](https://github.com/ZcashFoundation/zebra/pull/7041))

### Added

> - Metrics tracking mempool actions and size bucketed by weight ([#7019](https://github.com/ZcashFoundation/zebra/pull/7019))

Reviewer: Optional:

```suggestion
- Metrics tracking mempool actions and size bucketed by weight ([#7019](https://github.com/ZcashFoundation/zebra/pull/7019)) by @str4d
```

- Framework for upcoming in-place database format upgrades ([#7031](https://github.com/ZcashFoundation/zebra/pull/7031))

### Changed

- Deduplicate note commitment trees in non-finalized state ([#7218](https://github.com/ZcashFoundation/zebra/pull/7218), [#7239](https://github.com/ZcashFoundation/zebra/pull/7239))

### Fixed

- Enable miners running Zebra with Docker to set their address for mining rewards ([#7178](https://github.com/ZcashFoundation/zebra/pull/7178))
- Use default RPC port when running Zebra with Docker ([#7177](https://github.com/ZcashFoundation/zebra/pull/7177), [#7162](https://github.com/ZcashFoundation/zebra/pull/7162))
- Stop panicking on async task cancellation on shutdown in network and state futures ([#7219](https://github.com/ZcashFoundation/zebra/pull/7219))
- Remove redundant startup logs, fix progress bar number, order, and wording ([#7087](https://github.com/ZcashFoundation/zebra/pull/7087))
- Organize Docker `ENV` and `ARG` values based on their usage ([#7200](https://github.com/ZcashFoundation/zebra/pull/7200))

> -  Avoid blocking threads by awaiting proof verification results from rayon in async context ([#6887](https://github.com/ZcashFoundation/zebra/pull/6887))

Reviewer: Nitpick:

```suggestion
- Avoid blocking threads by awaiting proof verification results from rayon in async context ([#6887](https://github.com/ZcashFoundation/zebra/pull/6887))
```
